src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs

```diff
@@ -0,0 +1,179 @@
+		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+		{
+			Android.Database.ICursor _cursor = null;
+			Android.Content.ContentResolver _contentResolver = null;
+			List<Appointment> entriesList = new List<Appointment>();
+
+			if (options is null)
+			{
+				throw new ArgumentNullException();
+			}
+
+			Android.Net.Uri.Builder builder = Android.Provider.CalendarContract.Instances.ContentUri.BuildUpon();
+			Android.Content.ContentUris.AppendId(builder, rangeStart.ToUniversalTime().ToUnixTimeMilliseconds());
+			var rangeEnd = rangeStart + rangeLength;
+			Android.Content.ContentUris.AppendId(builder, rangeEnd.ToUniversalTime().ToUnixTimeMilliseconds());
+			Android.Net.Uri oUri = builder.Build();
+			// it is simply: {content://com.android.calendar/instances/when/1588275364371/1588880164371}
+
+			var androColumns = UWP2AndroColumnNames(options.FetchProperties);
+			// some 'system columns' columns, cannot be switched off
+			androColumns.Add(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			androColumns.Add("_id");
+			androColumns.Add(Android.Provider.CalendarContract.Instances.Begin);    // for sort
+			androColumns.Add(Android.Provider.CalendarContract.Instances.End);    // we need this, as Android sometimes has NULL duration, and it should be reconstructed from start/end
+
+			// string sortMode = Android.Provider.CalendarContract.EventsColumns.Dtstart + " ASC";
+			string sortMode = Android.Provider.CalendarContract.Instances.Begin + " ASC";
+			_contentResolver = Android.App.Application.Context.ContentResolver;
+			_cursor = _contentResolver.Query(oUri,
+									androColumns.ToArray(),  // columns in result
+									null,   // where
+									null,   // where params
+									sortMode);
+
+			if (_cursor is null || _cursor.IsAfterLast)
+			{
+				return entriesList;
+			}
+
+			if (!_cursor.MoveToFirst())
+			{
+				return entriesList;
+			}
+
+			// optimization
+			int _colAllDay, _colLocation, _colStartTime, _colSubject, _colOrganizer, _colDetails, _colCalId, _colId, _colEndTime;
+			_colAllDay = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
+			_colLocation = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
+			_colStartTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
+			_colEndTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
+			_colSubject = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
+			_colOrganizer = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+			_colDetails = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
+			_colCalId = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			_colId = _cursor.GetColumnIndex("_id");

```

Comment:
```suggestion
			var colAllDay = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
			var colLocation = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
			var colStartTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
			var colEndTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
			var colSubject = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
			var colOrganizer = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
			var colDetails = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
			var colCalId = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
			var colId = _cursor.GetColumnIndex("_id");

```

Reply:
This change is also done (maybe not exactly, as not to `var name = ...` but to `int name=...`).